Source/websocket/WebSocketLink.h

```diff
@@ -651,6 +651,20 @@ namespace Web {
             {
                 return (_protocol);
             }
+            inline bool IsProtocolSupported(const string& protocol)
```

Comment:
This is "higher level knowledge" and should not be inserted here.
 Do this check in the PluginServer, the server that really implements the sub protocolas and thus knows that it is possible.
here: https://github.com/rdkcentral/Thunder/blob/a42f71492333aafde27f9c431408721e7727c4f4/Source/websocket/WebSocketLink.h#L1000 the derviced class (e.g. the plugin server) gets an opportunity to "abort" an in-progress transition to a WebSocket. Check (and set) the used protocol, than in the parent (plugin server) and if the protocol is not supported, abort the upgrade through the AbortUpgrade method with a proper error.